doc/generate_config_rst.py

```diff
@@ -61,6 +61,8 @@ def print_entry(output, entry, name, required=False, depth=0):
 
 
 def has_subobjs(obj):
+    if not isinstance(obj, dict):
```

Comment:
```suggestion
    if not isinstance(obj, collections.abc.Mapping):
```

is more generic, although not in any way that matters here.